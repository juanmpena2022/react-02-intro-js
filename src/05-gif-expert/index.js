import React from "react";
import ReactDOM from "react-dom";
import { GifExpertApp } from "./05-gif-expert/components/GifExpertApp/GifExpertApp";

import "./index.css";

const divRoot = document.getElementById("root");
ReactDOM.render(<GifExpertApp />, divRoot);
