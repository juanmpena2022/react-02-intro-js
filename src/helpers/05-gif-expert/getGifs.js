export const getGifs = async (category) => {
  try {
    const url = `https://api.giphy.com/v1/gifs/search?api_key=6v0gqpBHPElle3D4YENpQCAHsVi55yKA&q=${encodeURI(
      category
    )}&limit=10`;
    const response = await fetch(url);
    const { data } = await response.json();

    const gifs = data.map((gif) => {
      return {
        id: gif.id,
        title: gif.title,
        url: gif.images?.downsized_medium.url,
      };
    });

    return gifs;
  } catch (error) {
    console.error(error);
  }
};
