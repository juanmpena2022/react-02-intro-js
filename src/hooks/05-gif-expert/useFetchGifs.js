import { useEffect, useState } from "react";
import { getGifs } from "../../helpers/05-gif-expert/getGifs";

export const useFetchGifs = (category) => {
  const [state, setState] = useState({
    data: [],
    loading: true,
  });

  useEffect(() => {
    getGifs(category).then((gifs) => {
      setState({
        data: gifs,
        loading: false,
      });
    });

    /*return () => {
        second
      }*/
  }, [category]);

  return state;
};
