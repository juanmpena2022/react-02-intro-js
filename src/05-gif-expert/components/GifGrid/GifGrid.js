import { useFetchGifs } from "../../../hooks/05-gif-expert/useFetchGifs";
import { GifGridItem } from "../GifGridItem/GifGridItem";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className="animate__animated animate__flash">{category}</h3>
      {loading && <p>Loading...</p>}
      <div className="card-grid">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
};
