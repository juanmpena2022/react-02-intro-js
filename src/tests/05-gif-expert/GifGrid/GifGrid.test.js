import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import { GifGrid } from "../../../05-gif-expert/components/GifGrid/GifGrid";
import { useFetchGifs } from "../../../hooks/05-gif-expert/useFetchGifs";
jest.mock("../../../hooks/05-gif-expert/useFetchGifs");

describe("pruebas del componente <GifGrid />", () => {
  const category = "One Punch";

  test("snapshot", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("debe mostrar items cuando se cargan imagenes", () => {
    const gifs = [
      {
        id: 1,
        url: "url",
        title: "title",
      },
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);
    const gifGridItem = wrapper.find('GifGridItem');
    const p = wrapper.find('p');

    expect(gifGridItem.exists()).toBe(true);
    expect(gifGridItem.length).toBe(gifs.length);
    expect(p.exists()).toBe(false);
  });
});
