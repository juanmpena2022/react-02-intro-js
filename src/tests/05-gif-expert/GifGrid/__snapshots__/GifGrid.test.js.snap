// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas del componente <GifGrid /> snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__flash"
  >
    One Punch
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
